### PR TITLE
remove tensor-array conversion in `Orientation`

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

### Description
non-breaking update to avoid tensor <-> ndarray conversion for the input image in `Orientation` transform

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_orientation.py [hunk 1]
```diff
@@ -110,14 +169,15 @@
...
+    def test_ornt(self, in_type, init_param, img, data_param, expected_data, expected_code):
+        img = in_type(img)
         ornt = Orientation(**init_param)
         res = ornt(img, **data_param)
         if not isinstance(res, tuple):
-            np.testing.assert_allclose(res, expected_data)
+            assert_allclose(res, expected_data, type_test=False)
             return
-        np.testing.assert_allclose(res[0], expected_data)
+        assert_allclose(res[0], expected_data, type_test=False)
```
Comment (rijobro) [inline tests/test_orientation.py:180]: if the tests are created such with `p(expected_data)`, then we can remove the `type_test`, right?

**Review by rijobro — COMMENTED** [review]
looks good to me, thanks!

File: monai/transforms/spatial/array.py [hunk 2]
```diff
@@ -310,21 +310,27 @@ def __call__(
...
+        if axes:
+            data_array = torch.flip(data_array, dims=axes) if _is_tensor else np.flip(data_array, axis=axes)  # type: ignore
+        full_transpose = np.arange(len(data_array.shape))
+        full_transpose[: len(spatial_ornt)] = np.argsort(spatial_ornt[:, 0])
+        data_array = (
+            data_array.permute(full_transpose.tolist()) if _is_tensor else data_array.transpose(full_transpose)  # type: ignore
+        )
         new_affine = to_affine_nd(affine_np, new_affine)
-        out, *_ = convert_to_dst_type(src=data_array_np, dst=data_array)
+        out = data_array.contiguous() if _is_tensor else np.ascontiguousarray(data_array)  # type: ignore
```
Comment (Nic-Ma) [inline monai/transforms/spatial/array.py:333]: I think you can simplify it to:
```py
out, *_ = convert_to_dst_type(src=data_array, dst=data_array)
```
The `ToTensor` and `ToNumpy` logic already have the `contiguous` conversion:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/type_conversion.py#L173
https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/type_conversion.py#L120

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me except for minor comment.

Thanks.

File: monai/transforms/spatial/array.py (hunk 2 above)
Comment (wyli) [reply monai/transforms/spatial/array.py]: thanks, I updated the PR and added logic to skip the transpose if not needed

**Comment by Nic-Ma** [comment]
Thanks for the quick update.
I think it looks good to me.
I am not very familiar with medical orientation computation, @ericspod @rijobro could you please help double confirm?

Thanks in advance.

**Pull request merged**